bin/commands/indexRestore.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const
+  fs = require('fs'),
+  ColorOutput = require('./colorOutput'),
+  getSdk = require('./getSdk');
+
+function importCollection (sdk, cout, dumpFiles) {
+  if (dumpFiles.length === 0) {
+    return null;
+  }
+
+  const bulkData = JSON.parse(fs.readFileSync(dumpFiles[0]));
```

Comment:
You can't put the entire data file in memory, and even less parse it into a JSON object: this method won't allow this CLI tool to handle even moderately large dumped collections.

You need to open a [readable stream](https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_createreadstream_path_options), and use it with some kind of buffer to send bulk imports to Kuzzle sequentially.

Reply:
Also, we may want to make the import buffer size configurable, to circumvent potentially strict limits put in the .kuzzlerc file by Kuzzle admins